### ISSUE TITLE
Warn on missing whitespace in CTE declaration

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#69]
I'd like to see a warning if there is missing whitespace after the `AS` in a CTE declaration, e.g.:
```
WITH
    count_audience_size AS( # <--- No space between AS and (
        SELECT
            user_id
        FROM
            table
    )

SELECT * FROM count_audience_size
```

**Comment by alanmcruickshank** [comment]
Interesting! I would have said there should be a single whitespace there. I think we've found the first of many contentious issues in sql linting.

I'm tempted in this case to look to formatting guides that already exist. According to the [dbt guidelines](https://github.com/fishtown-analytics/corp/blob/master/dbt_coding_conventions.md#ctes), the recommended format is:

```sql
with events as (

    ...

),

-- CTE comments go here
filtered_events as (

    ...

)

select * from filtered_events
```

I'm tempted to agree with the whitespace around `AS` and also the newline between `CTE`s in this case. @barrywhart thoughts?

**Comment by barrywhart** [comment]
I agree. I want the space after `AS` too. My wording was weird since I was saying "warn on <<bad behavior, i.e. MISSING whitespace>>" rather than "enforce good behavior, i.e. whitespace is present".
